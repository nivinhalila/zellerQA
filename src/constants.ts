import { ZellerAdmin, ZellerManager } from "./types";

export const ZELLER_ADMIN: ZellerAdmin = "ADMIN";
export const ZELLER_MANAGER: ZellerManager = "MANAGER";
