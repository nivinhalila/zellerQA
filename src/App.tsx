import React, { useState } from "react";
import styled from "styled-components";

import CustomerList from "./components/CustomerList";
import RoleList from "./components/RoleList";
import { ZellerRoles } from "./types";
import { ZELLER_ADMIN } from "./constants";

const HorizontalRule = styled.hr`
  border-top: 1px solid #e2e8f0;
`;

const Main = styled.main`
  padding: 2rem;
`;

function App() {
  const [role, setRole] = useState<ZellerRoles>(ZELLER_ADMIN);

  return (
    <Main>
      <RoleList role={role} setRole={setRole} />
      <HorizontalRule />
      <CustomerList role={role} />
      <HorizontalRule />
    </Main>
  );
}

export default App;
