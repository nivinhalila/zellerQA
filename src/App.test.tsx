import { MockedProvider } from "@apollo/client/testing";
import { render } from "@testing-library/react";

import App from "./App";
import { ZELLER_ADMIN } from "./constants";

test("renders app component", () => {
  expect.assertions(1);

  const { container } = render(
    <MockedProvider addTypename={false}>
      <App />
    </MockedProvider>
  );

  expect(container.firstChild).toMatchSnapshot();
});

test("sets admin radio as selected by default", () => {
  expect.assertions(1);

  const { getByTestId } = render(
    <MockedProvider addTypename={false}>
      <App />
    </MockedProvider>
  );

  const input = getByTestId("Admin") as HTMLInputElement;

  expect(input.checked).toEqual(true);
});
