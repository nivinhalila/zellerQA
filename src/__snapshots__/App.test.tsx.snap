// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app component 1`] = `
<main
  class="sc-dIsUp haQXTj"
>
  <section
    class="sc-iCoGMd gVpXrz"
  >
    <h1
      class="sc-fujyAs exdRzN"
    >
      User Types
    </h1>
    <label
      class="sc-kEqXSa gdqta-d"
      for="Admin"
    >
      <input
        checked=""
        class="sc-iqAclL XSvGs"
        data-testid="Admin"
        id="Admin"
        name="role"
        type="radio"
      />
      <p
        class="sc-bdnxRM bBxIvA"
      >
        Admin
      </p>
    </label>
    <label
      class="sc-kEqXSa dmdEjQ"
      for="Manager"
    >
      <input
        class="sc-iqAclL XSvGs"
        data-testid="Manager"
        id="Manager"
        name="role"
        type="radio"
      />
      <p
        class="sc-bdnxRM bBxIvA"
      >
        Manager
      </p>
    </label>
  </section>
  <hr
    class="sc-crzoAE jJogOo"
  />
  <section
    class="sc-iCoGMd gVpXrz"
  >
    <h1
      class="sc-fujyAs exdRzN"
    >
      ADMIN Users
    </h1>
    <p
      class="sc-bdnxRM sc-jrsJWt bBxIvA fbzdie"
    >
      Loading users...
    </p>
  </section>
  <hr
    class="sc-crzoAE jJogOo"
  />
</main>
`;
